OpenSim/Tools/CorrectionController.cpp

```diff
@@ -257,16 +255,26 @@ void CorrectionController::extendConnectToModel(Model& model)
     // create an actuator for each generalized coordinate in the model 
     // add these actuators to the model and set their indexes 
     const CoordinateSet& cs = _model->getCoordinateSet();
+    auto actuators = model.updComponentList<CoordinateActuator>();
+
     for(int i=0; i<cs.getSize(); i++) {
-        std::cout << " CorrectionController::extendConnectToModel(): " 
-                  <<  cs.get(i).getName()+"_corrector" << "  added " 
-                  << std::endl;
-        std::string name = cs.get(i).getName()+"_corrector";
-        CoordinateActuator *actuator = NULL;
-        if(_model->getForceSet().contains(name)){
-            actuator = (CoordinateActuator *)&_model->getForceSet().get(name);
+        const Coordinate& coord = cs[i];
+        std::string name = coord.getName() + "_corrector";
+
+        CoordinateActuator* actuator = nullptr;
+
+        for (auto& ca : actuators) {
+            if (ca.getName() == name) {
+                actuator = &ca;
+                break;
+            }
         }
-        else{
+
+        if(!actuator) {
+            std::cout << " CorrectionController::extendConnectToModel(): "
```

Comment:
This comment seems to be premature, and if something throws on L278–289, it would seem like the Controller had been added successfully. I suggest either changing "...added" to "adding...", or (perhaps better) move this comment down, after `addActuator(*actuator);`.

Reply:
Good point.